app/uk/gov/hmrc/eusubsidycompliance/connectors/EisConnector.scala

```diff
@@ -49,19 +49,19 @@ class EisConnector @Inject() (
   private val amendSubsidyPath = "scp/amendundertakingsubsidyusage/v1"
   private val retrieveSubsidyPath = "scp/getundertakingtransactions/v1"
 
-  def retrieveUndertaking(eori: EORI)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Undertaking] = {
+  def retrieveUndertaking(eori: EORI)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[Undertaking]] = {
 
     import uk.gov.hmrc.eusubsidycompliance.models.json.digital.{retrieveUndertakingEORIWrites, undertakingReads}
 
     val eisTokenKey = "eis.token.scp04"
-    desPost[EORI, Undertaking](
+    desPost[EORI, Option[Undertaking]](
       s"$eisURL/$retrieveUndertakingPath",
       eori,
       eisTokenKey
     )(implicitly, implicitly, addHeaders, implicitly).recover {
       case e: EisBadResponseException if e.code == EisParamValue("107") =>
         logger.info(s"No undertaking found for $eori")
-        throw UpstreamErrorResponse.apply("undertaking not found", 404)
+        None
```

Comment:
Just checking that the backend service will continue to return a HTTP 404 for this case? (The frontend changes suggest this has changed to a 200 but I could be misreading it).